dcos/dcos-ui
# [1.10] DCOS-18055: Improvements for deployment behavior in Catalog

---
⚠️  _This branch depends on changes that are going be introduced with #2413. For a list of changes please see: https://github.com/dcos/dcos-ui/compare/orlandohohmeier/1.10/chore/remove-beta-opt-in...mat/review-and-run_ 

---

On **catalog > package detail** page.
- Consolidates "CONFIGURE" and "DEPLOY" buttons into a single "REVIEW & RUN" button.
- Opens review screen first with options to edit configuration or run service

**To better understand these changes, walk through the commits from top to bottom and read the commit messages.**

**To test**:
- Run Service without editing configuration. Attempt to run service again with same name and witness failure failure.
- Edit configuration for a package to change name and run service. Ensure service is then running.
- Try to run artifactory package without editing. Should fail, show option to edit config. Click option to edit and witness errors in the form.


**Checklist**
